test_api.py

```diff
@@ -1795,61 +1794,65 @@ def setUp(self):
+            self._assert_event_is_valid(self.app.event_producer, self.host_to_delete, self.timestamp)
+            self._check_hosts_are_deleted((self.host_to_delete.id,))
 
-    def test_create_then_delete_with_request_id(self, emit_event):
-        request_id = generate_uuid()
-        header = {"x-rh-insights-request-id": request_id}
-        self._delete(header=header)
+    def test_create_then_delete_with_request_id(self):
+        with self.app.app_context():
+            request_id = generate_uuid()
+            header = {"x-rh-insights-request-id": request_id}
+            self._delete(header=header)
+
+            self._assert_event_is_valid(self.app.event_producer, self.host_to_delete, self.timestamp)
 
-        emit_event.assert_called_once()
-        event = emitted_event(emit_event.mock_calls[0])
-        self._assert_event_is_valid(event, self.host_to_delete, self.timestamp)
-        self.assertEqual(request_id, event.value["request_id"])
+            event = json.loads(self.app.event_producer.event)
+            self.assertEqual(request_id, event["request_id"])
 
-    def test_create_then_delete_without_request_id(self, emit_event):
-        self._check_hosts_are_present((self.host_to_delete.id,))
-        self._delete(header=None)
+    def test_create_then_delete_without_request_id(self):
+        with self.app.app_context():
+            self._check_hosts_are_present((self.host_to_delete.id,))
+            self._delete(header=None)
+
+            self._assert_event_is_valid(self.app.event_producer, self.host_to_delete, self.timestamp)
 
-        emit_event.assert_called_once()
-        event = emitted_event(emit_event.mock_calls[0])
-        self._assert_event_is_valid(event, self.host_to_delete, self.timestamp)
-        self.assertEqual("-1", event.value["request_id"])
+            event = json.loads(self.app.event_producer.event)
+            self.assertEqual("-1", event["request_id"])
 
 
-@patch("lib.host_delete.emit_event")
 class DeleteHostsRaceConditionTestCase(PreCreatedHostsBaseTestCase):
+    def setUp(self):
+        super().setUp()
+        self.event_producer = mock.Mock()
+
     class DeleteHostsMock:
         @classmethod
         def create_mock(cls, hosts_ids_to_delete):
-            def _constructor(select_query):
-                return cls(hosts_ids_to_delete, select_query)
+            def _constructor(select_query, request_id, event_producer):
+                return cls(hosts_ids_to_delete, select_query, event_producer)
 
             return _constructor
 
-        def __init__(self, host_ids_to_delete, original_query):
+        def __init__(self, host_ids_to_delete, original_query, event_producer):
             self.host_ids_to_delete = host_ids_to_delete
-            self.original_query = delete_hosts(original_query)
+            with self.app.app_context():
+                self.original_query = delete_hosts(original_query, "-1", self.app.event_producer)
```

Comment:
You created a different [_event_producer_](https://github.com/RedHatInsights/insights-host-inventory/blob/e1c4e8373c802569bfc767358d7cfa74d397c2b5/test_api.py#L1842) in _setUp_, but here you are passing in _self.app.event_producer_. But even if you used _self.event_producer_ here, it still wouldn’t work. Here, _self_ refers to a _DeleteHostsMock_ class instance, but the event producer is attached to _DeleteHostsRaceConditionTestCase_.

Just use the _event_producer_ you receive as an argument. It’s passed in by the caller that uses _current_app.event_producer_. That is the same object as _self.app.event_producer_ here in the tests.

```suggestion
                self.original_query = delete_hosts(original_query, "-1", event_producer)
```